src/current/v23.2/changefeed-messages.md

```diff
@@ -191,70 +191,134 @@ CREATE CHANGEFEED FOR TABLE rides INTO 'external://kafka' WITH diff, envelope=wr
     {"city": "washington dc", "creation_time": "2019-01-02T03:04:05", "current_location": "85551 Moore Mountains Apt. 47", "ext": {"color": "red"}, "id": "d3b37607-1e9f-4e25-b772-efb9374b08e3", "owner_id": "4f26b516-f13f-4136-83e1-2ea1ae151c20", "status": "available", "type": "skateboard"}
     ~~~
 
-## Ordering guarantees
+## Ordering and delivery guarantees
 
-- In most cases, each version of a row will be emitted once. However, some infrequent conditions (e.g., node failures, network partitions) will cause them to be repeated. This gives our changefeeds an **at-least-once delivery guarantee**.
+Changefeeds provide the following guarantees for message delivery to changefeed sinks:
 
-- Once a row has been emitted with some timestamp, no previously unseen versions of that row will be emitted with a lower timestamp. That is, you will never see a _new_ change for that row at an earlier timestamp.
+- [Per-key ordering](#per-key-ordering)
+- [At-least-once delivery](#at-least-once-delivery)
 
-    For example, if you ran the following:
+{{site.data.alerts.callout_info}}
+Changefeeds do not support total message ordering or transactional ordering of messages.
+{{site.data.alerts.end}}
+
+### Per-key ordering
+
+Changefeeds provide a _per-key ordering guarantee_ for messages emitted to the sink. Once the changefeed has emitted a row with a timestamp, the changefeed will not emit any previously unseen versions of that row with a lower timestamp. Therefore, you will never receive a **new** change for that row at an earlier timestamp.
+
+For example, a changefeed can emit updates to rows A at timestamp T1, B at T2, and C at T3 in any order.
 
+When there are updates to rows A at T1, B at T2, and A at T3, the changefeed will always emit A at T3 **after** A at T1. However, A at T3 could precede B at T2. This is because there is no timestamp ordering between keys.
```

Comment:
> However, A at T3 could precede B at T2.

We can also emit `A@T3` _after_ `B@T2`, right? Might be worth calling this out too, as it underscores the fact that we only order within the same key.

Reply:
I have added this in, it's a good idea that helps to solidify the point here.